---
title: "Analysis of CRC Spatial Transcriptomics data from 10X: Normalization and Clustering"
author: 'Alberto Valdeolivas Urbelz (PS-Biomics&Path)'
date: '`r format(Sys.time(), "%d %B, %Y")`'
output:
  html_document:
    code_folding: hide
    self_contained: true
    number_sections: yes
    toc: true
    toc_depth: 2
    toc_float:
      collapsed: false
      smooth_scroll: false
    theme: spacelab
  pdf_document:
    number_sections: yes
    toc: yes
    toc_depth: 3
params:
  data_directory : "/projects/site/pred/SpatialOmics/Experiment_CRC_AllSamples/"
  analysis_name : "analysis_alberto/"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Introduction

The present script takes the the Spatial Omics data generated with 10X 
technology on a set of CRC samples and performs normalization and 
clustering steps. In addition, we find the markers for every cluster and 
characterize them. 

```{r, warning=FALSE,message=FALSE}
library(ggplot2)
library(vctrs)
library(patchwork, lib.loc = "/apps/rocs/2020.08/cascadelake/software/R/4.1.2-foss-2020a/lib64/R/library")
library(Seurat)
library(dplyr)
library(stringr)
library(purrr)
library(readr)
library(tidyr)
library(ggplot2)
#library(tidyverse)
library(kableExtra)
library(gridExtra)
library(cowplot)
library(msigdbr)
library(clusterProfiler)
# library(EnhancedVolcano)
source(file = "WrapperFunction/SeuratWrappers.R")

data_directory <-  params$data_directory
analysis_name <- params$analysis_name

input_files <- "IntermediaryFiles/"
output_names <- "SeuratList_Clusters_Res05.rds"


data_directory_counts <- paste0(data_directory, "Fastq_Merged")
counts_directory <- "/outs"
results_directory <- paste0(data_directory, analysis_name, input_files)
```

```{r, message=FALSE, warning=FALSE}
datasets <- 
  list.dirs(path = data_directory_counts, recursive = FALSE, full.names = TRUE) %>% 
  str_subset("Count_") %>%
  paste0(counts_directory)

sample_names <- 
  list.dirs(path = data_directory_counts, recursive = FALSE, full.names = FALSE) %>% 
  str_subset("Count_") %>%
  str_replace("Count_", "") 
```

```{r, message=FALSE, warning=FALSE}
## Generating Seurat Objects 
seurat_objects <- 
  get.seurat.objects(datasets,sample_names)

seurat_objects <- 
  get.qc.mt.percentage(seurat_objects, mypattern = "^MT-")

## We filter our samples based on the results of the QC. Non strict threshold, 
## we want to keep as many spots as possible.

seurat_objects_filter <-seurat_objects %>% 
  map(function(x) {
    x %>%
    subset(subset  = percent.mt < 50  & nCount_Spatial > 500 & nCount_Spatial < 45000)  
  })
```


```{r, message=FALSE, warning=FALSE}
seurat_objects <- normalize.cluster(seurat_objects_filter, min.dist = 0.1, 
  spread = 0.5, resolution = 0.5)

umap_spatial_plots <- get.umap.spatialClusters(seurat_objects, 
   label.size = 3,  pt.size.factor = 1.75)

spatial_dim_plot_ident <- lapply(seurat_objects, function(x){
  SpatialDimPlot(x, cells.highlight = CellsByIdentities(object = x), 
  facet.highlight = TRUE, ncol = 2, pt.size.factor = 2.5,
  cols.highlight = c("#8b0023", "#ffffff00"), stroke = 0)
})

all_markers <- 
  get.all.markers(seurat_objects, logfc.threshold = 0.5, return.thresh = 0.01, 
  only.pos = TRUE) 
all_markers_table <- get.markers.table(all_markers, nr_markers = 2)

write.all.markers(all_markers, 
  path = "IntermediaryFiles/ClusteringResults/Resolution05/")
write.loupe.file(seurat_objects, 
  path = "IntermediaryFiles/ClusteringResults/Resolution05/") 

background_genes <- lapply(seurat_objects, function(x) VariableFeatures(x))

my_categories <- c("C7", "C8")
my_sub_categories <- 
   c("CP:KEGG", "CP:REACTOME", "CP:BIOCARTA")
all_gene_sets <- msigdbr(species = "Homo sapiens")

gene_sets_selected <- 
  dplyr::filter(all_gene_sets, 
    (gs_subcat %in% my_sub_categories)) #  |  (gs_cat %in% my_categories))

msigdbr_t2g <- gene_sets_selected %>% 
   dplyr::select(gs_name, gene_symbol) %>% 
  as.data.frame()

enrichment_results <- get.enrichment.results.cp(all_markers, background_genes, 
   pvalueCutoff = 0.01,  pAdjustMethod = "fdr", TERM2GENE = msigdbr_t2g)
```

# Results for the different Samples {.tabset}

```{r, results = 'asis', message=FALSE, dpi=80, warning=FALSE}
for ( sample in names(seurat_objects )){
  
  cat( "## Sample : ", sample , "\n\n" )
  
  cat( "Showing sample : ", sample , "\n\n")
  
  cat( "### Clusters and their molecular characterization \n\n")
  
  cat( "**Umap and Spatial Plot of clusters** \n\n")

  print( umap_spatial_plots[[sample]])
  
  cat( "\n\n")
  
  print(spatial_dim_plot_ident[[sample]])
  
  cat( "\n\n")
  
  cat( "**Table containing some of the top markers of the clusters** \n\n")
  
  print( all_markers_table[[sample]] %>% 
      kbl() %>% kable_styling())
  
  cat( "\n\n")
  
  top5 <- all_markers[[sample]] %>% 
    group_by(cluster) %>% 
    top_n(n = 5, wt = avg_log2FC)

  cat( "**Heatmap showing the markers of the different clusters ** \n\n")
  
  print(DoHeatmap(seurat_objects[[sample]], features = top5$gene) + NoLegend())
  
  cat( "\n\n")
  
  cat( "**Spatial and Violin plots showing the expression of the markers of different clusters ** \n\n")
  
  for (cluster_nr in levels(Idents(seurat_objects[[sample]]))){
    cat( "Cluster : ", cluster_nr , "\n\n")
    genes_cluster <- all_markers_table[[sample]] %>%
      dplyr::filter(cluster == as.numeric(cluster_nr)) %>%
      dplyr::pull(gene)
    print(SpatialFeaturePlot(seurat_objects[[sample]], 
      features = genes_cluster, 
      pt.size.factor = 2.25, alpha = c(0.1,1),ncol=2))
    cat( "\n\n")
    print(VlnPlot(seurat_objects[[sample]], 
                  features = genes_cluster, ncol=2))
    cat( "\n\n")
    
  }

  cat( "\n\n")
  
  cat( "**Enrichment Results:**  The dashed line is the statistical treshold for a significant enrichment \n\n")
  
  print(get.enrichment.barplots.cp(enrichment_results[[sample]]))
  
  cat( "\n\n")
  
}
```

# Saving the results

```{r, eval=FALSE}
saveRDS(seurat_objects, file = paste0(results_directory, output_names))
```


# Conclusion 


# Session Info Details

```{r, echo=FALSE, eval=TRUE}
sessionInfo()
```